core/prop.cpp

```diff
@@ -58,4 +52,50 @@ Property::Property(Property & prop, TermTranslator & tt)
 
 Property::~Property() {}
 
+void Property::initialize()
+{
+  if (ts_.only_curr(prop_)) {
+    // nothing to do
+    return;
+  }
+
+  // otherwise, need to make sure prop_ is over state vars
+  logger.log(1,
+             "Got next state or input variables in property. Generating a "
+             "monitor state.");
+
+  Term monitor;
+  size_t id = 0;
+  while (true) {
+    try {
+      monitor = ts_.make_statevar("_monitor_" + std::to_string(id),
+                                  ts_.make_sort(BOOL));
+      break;
+    }
+    catch (SmtException & e) {
```

Comment:
I am not very familiar with the `smt-switch` code: is it guaranteed that an `SmtException` will **always** be thrown when `ts_.make_statevar(...)` fails, regardless of the cause of failure? (to avoid getting stuck in the infinite loop)

Reply:
Good question -- an `SmtException` is the parent class for all exceptions thrown by `smt-switch`, but it is possible that some other type of exception is thrown in the underlying solver and is not caught correctly by smt-switch. In that case, I believe we still wouldn't be stuck in the infinite loop because we'd get an uncaught exception.

Reply:
I see, that makes sense!